bin/ruby-install

```diff
@@ -18,6 +18,10 @@ fi
 
 init || exit $?
 
+if [[ ! -w "$install_dir" ]]; then
```

Comment:
I just realized an edge-case. If `$install_dir` doesn't exist yet, such as when running `ruby-install` for the first time on a new system that doesn't have `/opt/rubies` or `~/.rubies` yet, it will consider the `$install_dir` non-writable. We should also check for it's existence *and* if it's not-writable.

```shell
if [[ -d "$install_dir" ]] && [[ ! -w "$install_dir" ]]; then
  ...
fi
```

Reply:
This might cause problems if the user specifies a really long `--install-dir` who's parent directories do not exist yet.

```
ruby-instlal --install-dir `$HOME/does/not/exist/yet` ruby
```

We could move this logic into `share/ruby-install/functions.sh` `pre_install` which already does a `mkdir -p "${install_dir%/*}" || return $?`.

Reply:
oh yes, that was close!!!

Reply:
there's an issue I think with this, for example if `install_dir` is of this form : `/this/part/exists/this/part/exists/not`, we should recursively (iteratively) return backwards to check for the existing part, before testing if it was writable or not; then we proceed with our normal operation

Reply:
Or we could take advantage of [`mkdir -p`](https://github.com/postmodern/ruby-install/blob/61a627e436601c847cfa850c76cd3433d5e465b6/share/ruby-install/functions.sh#L11) which recursively creates the directories.

Reply:
it won't create them if we don't have write permissions over `/this/part/exists/`

Reply:
If the `mkdir -p` fails because the last existing sub-directory isn't writable then the command will fail, which could be detected, an error message printed, and an error code returned.